support.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"strings"
+)
+
+// Returns the Syntax Dictionary that maps keywords to functions that handle the processing of their output.
+// The functions will each accept a string, a string slice, and an integer
+// The string will be the line from out.txt that calls the custom syntax, this is so any necessary parameters can be extracted out.
+// The string slice is the slice containing all of the lines output to stdout
+// The integer will be the current position in the above slice that the grader is in
+// Furthermore, the functions will return two integer values.
+// The first integer represents the result score, which will hold either 1 or 0.
+// The second integer will be the new position in the slice containing stdout
+func initSyntaxDictionary() map[string]func(string, []string, int) (string, int, []string) {
+	SyntaxDictionary := make(map[string]func(string, []string, int) (string, int, []string))
+
+	SyntaxDictionary["menu"] = func(menuCall string, StdOutput []string, startPos int) (string, int, []string) {
```

Comment:
would it make more sense to have this function moved into the global namespace? in golang, functions can be stored as variables:

```go
// declare a type for the functions in the array
type SDFunc func(string, []string, int) (string, int, []string)
func simpleSD(s string, ss []string, i int) (string, int, []string) {
  return s, i, ss
}

SD := make(map[string]SDFunc)
SD["simple"] = simpleSD
```

I _think_ this works, not sure though.

Also not blocking.

Reply:
I agree, this makes more sense. I'll work on an implementation.